Source/ChocolateyGui.Shared/Models/SemanticVersion.cs

```diff
@@ -0,0 +1,337 @@
+
+            var empty = string.IsNullOrEmpty(SpecialVersion);
+            var otherEmpty = string.IsNullOrEmpty(other.SpecialVersion);
+            if (empty && otherEmpty)
+            {
+                return 0;
+            }
+
+            if (empty)
+            {
+                return 1;
+            }
+
+            if (otherEmpty)
+            {
+                return -1;
+            }
+
+            return StringComparer.OrdinalIgnoreCase.Compare(SpecialVersion, other.SpecialVersion);
+        }
+
+        public static bool operator ==(SemanticVersion version1, SemanticVersion version2)
+        {
+            if (ReferenceEquals(version1, null))
+            {
+                return ReferenceEquals(version2, null);
+            }
+
+            return version1.Equals(version2);
+        }
+
+        public static bool operator !=(SemanticVersion version1, SemanticVersion version2)
+        {
+            return !(version1 == version2);
+        }
+
+        public static bool operator <(SemanticVersion version1, SemanticVersion version2)
+        {
+            if (version1 == null)
+            {
+                throw new ArgumentNullException(nameof(version1));
+            }
+
+            return version1.CompareTo(version2) < 0;
+        }
+
+        public static bool operator <=(SemanticVersion version1, SemanticVersion version2)
+        {
+            return (version1 == version2) || (version1 < version2);
+        }
+
+        public static bool operator >(SemanticVersion version1, SemanticVersion version2)
+        {
+            if (version1 == null)
+            {
+                throw new ArgumentNullException(nameof(version1));
+            }
+
+            return version2 < version1;
+        }
+
+        public static bool operator >=(SemanticVersion version1, SemanticVersion version2)
+        {
+            return (version1 == version2) || (version1 > version2);
+        }
+
+        public override string ToString()
+        {
+            return _originalString;
+        }
+
+        public bool Equals(SemanticVersion other)
+        {
+            return !ReferenceEquals(null, other) &&
+                   Version.Equals(other.Version) &&
+                   SpecialVersion.Equals(other.SpecialVersion, StringComparison.OrdinalIgnoreCase);
+        }
+
+        public override bool Equals(object obj)
+        {
+            var semVer = obj as SemanticVersion;
+            return !ReferenceEquals(null, semVer) && Equals(semVer);
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = Version.GetHashCode();
+            if (SpecialVersion != null)
+            {
+#pragma warning disable SA1407 // Arithmetic expressions must declare precedence
+                hashCode = hashCode * 4567 + SpecialVersion.GetHashCode();
+#pragma warning restore SA1407 // Arithmetic expressions must declare precedence
+            }
+
+            return hashCode;
+        }
+    }
+}
```

Comment:
Nope, this was because I pulled chocolatey.lib out of chocolateygui (so it's only the in the subprocess). However, there's 0 reason I can't just put it back, especially since I'm just using a model.